src/lib/viewers/controls/slider/SliderControl.tsx

```diff
@@ -1,42 +1,151 @@
 import React from 'react';
 import classNames from 'classnames';
 import noop from 'lodash/noop';
+import { decodeKeydown } from '../../../util';
 import './SliderControl.scss';
 
-export type Props = Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange'> & {
+export type Props = React.HTMLAttributes<Ref> & {
     className?: string;
-    onChange?: (value: number) => void;
+    max?: number;
+    min?: number;
+    onMove?: (value: number, position: number, width: number) => void;
+    onUpdate?: (value: number) => void;
+    step?: number;
+    title: string;
     track?: string;
+    value: number;
 };
 
-export type Ref = HTMLInputElement;
+export type Ref = HTMLDivElement;
 
-function SliderControl({ className, onChange = noop, track, ...rest }: Props, ref: React.Ref<Ref>): JSX.Element {
-    const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>): void => {
-        onChange(parseFloat(target.value));
+export default function SliderControl({
+    className,
+    max = 100,
+    min = 0,
+    onMove = noop,
+    onUpdate = noop,
+    step = 1,
+    title,
+    track,
+    value,
+    ...rest
+}: Props): JSX.Element {
+    const [isScrubbing, setIsScrubbing] = React.useState(false);
+    const sliderElRef = React.useRef<Ref>(null);
+
+    const getPosition = React.useCallback((pageX: number) => {
+        const { current: sliderEl } = sliderElRef;
+
+        if (!sliderEl) return 0;
+
+        const { left: sliderLeft, width: sliderWidth } = sliderEl.getBoundingClientRect();
+        return Math.max(0, Math.min(pageX - sliderLeft, sliderWidth));
+    }, []);
+
+    const getPositionValue = React.useCallback(
+        (pageX: number) => {
+            const { current: sliderEl } = sliderElRef;
+
+            if (!sliderEl) return 0;
+
+            const { width: sliderWidth } = sliderEl.getBoundingClientRect();
+            const newValue = (getPosition(pageX) / sliderWidth) * max;
```

Comment:
Do we need to worry about decimal values here?

Reply:
I think we'd want to let the consuming component handle rounding to avoid making assumptions about its use. Videos can have durations defined as decimal values, for whatever reason, for example.